src/modules/Dropdown/Dropdown.js

```diff
@@ -188,14 +180,35 @@ export default class Dropdown extends Component {
 
   componentWillReceiveProps(nextProps) {
     super.componentWillReceiveProps(nextProps)
-    debug.groupCollapsed('componentDidUpdate()')
+    debug.groupCollapsed('componentWillReceiveProps()')
     // TODO objectDiff still runs in prod, stop it
     debug('changed props:', objectDiff(nextProps, this.props))
-    debug.groupEnd()
 
-    if (nextProps.value !== this.state.value) {
+    if (process.env.NODE_ENV !== 'production') {
+      // in development, validate value type matches dropdown type
+      const isNextValueArray = Array.isArray(nextProps.value)
+
+      /* eslint-disable no-console */
+      if (nextProps.multiple && !isNextValueArray) {
+        console.error(
+          'Dropdown `value` must be an array when `multiple` is set.' +
+          ` Received type: \`${Object.prototype.toString.call(nextProps.value)}\`.`,
+        )
+      } else if (!nextProps.multiple && isNextValueArray) {
+        console.error(
+          'Dropdown `value` must not be an array when `multiple` is not set.' +
```

Comment:
Double negation takes a re-read to understand.

`Dropdown 'value' can only been an array when multiple is set`